### DISCUSSION TITLE
when  will rustdesk support raspbian with desktop version?

### DISCUSSION THREAD

**Discussion by rinex20** [rustdesk/rustdesk#697] [General]
when  will rustdesk support raspbian with desktop version?

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/175

**Comment by rustdesk** [comment]
When will some one donate us a PI? Hate QEMU. :)

**Comment by BlackDragon-B** [comment]
> When will some one donate us a PI? Hate QEMU. :)

When you buy one yourself. No one is gonna buy you one, guess you have to cope with QEMU.

**Comment by rustdesk** [comment]
@rinex20 Thanks for you willing to donate a PI. I just asked around, one of our contributors has a B4 8G PI, he did some test last month, but RustDesk runs slowly on his PI, not deeply investigate yet. @Heap-Hop

**Reply by rinex20** [reply]
Most of users, include me I think we need just a server on raspbian.
we don’t need access other desktop from rpi.

for this opinion, can we reduce the app’s function to optimized the speed in raspberry pi.

**Reply by rustdesk** [reply]
It is not practical for us to invest time on raspberry pi for the time being. You can only rely on the community.

**Comment by Heap-Hop** [comment]
Quick way to build RustDesk on Raspberry 
(32bit armhf only).

### 1. Prepare arm environment
Raspberry Pi4 Raspberry OS **32bit**
at lease RAM>=4GB

### 2. Install all deps
```
sudo apt install -y g++ gcc git curl wget nasm yasm libgtk-3-dev clang libxcb-randr0-dev libxdo-dev libxfixes-dev libxcb-shape0-dev libxcb-xfixes0-dev libasound2-dev libpulse-dev cmake libvpx-dev libopus-dev libjpeg-dev libturbojpeg-dev
```
NOTICE:
We must install `libyuv-dev` and `libyuv0` manually, even though they are already exist on apt source: http://raspbian.raspberrypi.org/raspbian/pool/main/liby/libyuv/

```
# download 
# libyuv0_0.0_git20220427.de71c67-1_armhf.deb
# libyuv-dev_0.0_git20220427.de71c67-1_armhf.deb

# and install
sudo dpkg -i libyuv0*.deb
sudo dpkg -i libyuv-dev*.deb
```


if you do that by cmd `apt` like:
```
apt install libyuv-dev libyuv0
# or 
apt search libyuv-dev libyuv0
```
doesn't work in current Raspberry OS, I don't know why.


### 3. cargo run
```
# make a fake VCPKG dir and link libs
mkdir -p $HOME/vcpkg/installed/arm-linux
export VCPKG_ROOT=$HOME/vcpkg

ln -s /usr/include $HOME/vcpkg/installed/arm-linux/include
ln -s /usr/lib/arm-linux-gnueabihf $HOME/vcpkg/installed/arm-linux/lib


git clone https://github.com/rustdesk/rustdesk
cd rustdesk

wget https://raw.githubusercontent.com/c-smile/sciter-sdk/master/bin.lnx/arm32/libsciter-gtk.so
mkdir -p target/debug
mv libsciter-gtk.so target/debug

cargo run

# for Pi4 8G, about 10~15mins
```
![pi1](https://user-images.githubusercontent.com/62206297/171881601-7e6bb0eb-31d3-491b-9600-3988564c0296.png)

### NOTICE:
We haven't done ANY optimization for arm yet, maybe there are some unknown problems.

**Contributions welcome!**

*A known issue is that the UI cannot be translated, so it only works well in English.

**Reply by potterpg** [reply]
I followed this

Had to download different versions of the libs...
libyuv0_0.0_git20220617.e906ba9-1_armhf.deb
libyuv-dev_0.0_git20220617.e906ba9-1_armhf.deb

 and finally got a 
`Cargo:  command not found`

So I'm missing something? a quick search suggests maybe I need to install rust package manager?

Installed that, now it runs, but I get an error...

`
error: failed to parse manifest at `/home/pi/rustdesk/Cargo.toml`

`
Caused by:
  invalid type: unit variant, expected string only for key `profile.release.strip`
`

Maybe I'm missing something else?

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/releases/download/1.1.9/rustdesk-1.1.9-raspberry-armhf.deb
